readthedocs/rtd_tests/tests/test_oauth.py

```diff
@@ -1,24 +1,874 @@
+            # "comments_url": "https://api.github.com/repos/octocat/Hello-World/commits/6dcb09b5b57875f334f61aebed695e2e4193db5e/comments",
+            "commit": {
+                "url": f"https://api.github.com/repos/octocat/Hello-World/git/commits/{commit}",
+                "author": {
+                    "name": "Monalisa Octocat",
+                    "email": "mona@github.com",
+                    "date": "2011-04-14T16:00:49Z",
+                },
+                "committer": {
+                    "name": "Monalisa Octocat",
+                    "email": "mona@github.com",
+                    "date": "2011-04-14T16:00:49Z",
+                },
+                "message": "Fix all the bugs",
+                "tree": {
+                    "url": f"https://api.github.com/repos/user/repo/tree/{commit}",
+                    "sha": commit,
+                },
+                "comment_count": 0,
+            },
+            "author": {
+                "login": "octocat",
+                "id": 1,
+                "avatar_url": "https://github.com/images/error/octocat_happy.gif",
+                "gravatar_id": "",
+                "url": "https://api.github.com/users/octocat",
+                "html_url": "https://github.com/octocat",
+                "type": "User",
+                "site_admin": False,
+            },
+            "committer": {
+                "login": "octocat",
+                "id": 1,
+                "avatar_url": "https://github.com/images/error/octocat_happy.gif",
+                "gravatar_id": "",
+                "url": "https://api.github.com/users/octocat",
+                "html_url": "https://github.com/octocat",
+                "type": "User",
+                "site_admin": False,
+            },
+            "parents": [
+                {
+                    "url": "https://api.github.com/repos/user/repo/commits/6dcb09b5b57875f334f61aebed695e2e4193db5e",
+                    "sha": "6dcb09b5b57875f334f61aebed695e2e4193db5e",
+                }
+            ],
+            "stats": {"additions": 104, "deletions": 4, "total": 108},
+            "files": [],
+        }
+        self._merge_dicts(default, kwargs)
+        return default
+
+    def test_for_project(self):
+        services = list(GitHubAppService.for_project(self.project))
+        assert len(services) == 1
+        service = services[0]
+        assert service.installation == self.installation
+
+        services = list(GitHubAppService.for_project(self.project_with_org))
+        assert len(services) == 1
+        service = services[0]
+        assert service.installation == self.organization_installation
+
+    @requests_mock.Mocker(kw="request")
+    def test_for_user(self, request):
+        new_installation_id = 3333
+        assert not GitHubAppInstallation.objects.filter(
+            installation_id=new_installation_id
+        ).exists()
+        request.get(
+            "https://api.github.com/user/installations",
```

Comment:
Is there any reason why we are not using `self.api_url` here as in the other tests?

Reply:
API url includes the port `:443`, as that's how pygithub uses the API, but for getting the installations we use our OAuth2 client (as we need to make use of the oauth2 token for that action)

https://github.com/readthedocs/readthedocs.org/blob/0518dab7186e209c1bbb7485f7c0d393dfd10389/readthedocs/oauth/services/githubapp.py#L107-L108